packages/rpc-core/src/rpc-methods/getVoteAccounts.ts

```diff
@@ -0,0 +1,47 @@
+import { Base58EncodedAddress } from '@solana/keys';
+import { Commitment, U64UnsafeBeyond2Pow53Minus1 } from './common';
+
+type Epoch = U64UnsafeBeyond2Pow53Minus1;
+type Credits = U64UnsafeBeyond2Pow53Minus1;
+type PreviousCredits = U64UnsafeBeyond2Pow53Minus1;
+
+type EpochCredit = [Epoch, Credits, PreviousCredits];
+
+type VoteAccount = Readonly<{
+    /** Vote account address */
+    votePubkey: Base58EncodedAddress;
+    /** Validator identity */
+    nodePubkey: Base58EncodedAddress;
+    /** the stake, in lamports, delegated to this vote account and active in this epoch */
+    activatedStake: U64UnsafeBeyond2Pow53Minus1;
+    /** whether the vote account is staked for this epoch */
+    epochVoteAccount: boolean;
+    /** percentage (0-100) of rewards payout owed to the vote account */
+    commission: number;
+    /** Most recent slot voted on by this vote account */
+    lastVote: U64UnsafeBeyond2Pow53Minus1;
+    /** Latest history of earned credits for up to five epochs */
+    epochCredits: [EpochCredit];
+    /** Current root slot for this vote account */
+    rootSlot: U64UnsafeBeyond2Pow53Minus1;
+}>;
+
+type GetVoteAccountsApiResponse = Readonly<{
+    current: [VoteAccount];
+    delinquent: [VoteAccount];
+}>;
+
+type GetVoteAccountsConfig = Readonly<{
+    commitment?: Commitment;
+    /** Only return results for this validator vote address */
+    votePubkey?: Base58EncodedAddress;
+    /** Do not filter out delinquent validators with no stake */
+    keepUnstakedDelinquents?: boolean;
+    /** Specify the number of slots behind the tip that a validator must fall to be considered delinquent. **NOTE:** For the sake of consistency between ecosystem products, _it is **not** recommended that this argument be specified._ */
+    delinquentSlotDistance?: U64UnsafeBeyond2Pow53Minus1;
+}>;
+
+export interface GetVoteAccountsApi {
+    /** Returns the account info and associated stake for all the voting accounts in the current bank. */
+    getVoteAccounts(config?: GetVoteAccountsConfig): GetVoteAccountsApiResponse;
```

Comment:
Do you think we can get the type of the `votePubkey` to flow through now?

```suggestion
    getVoteAccounts<TVoteAccount>(
      config?: GetVoteAccountsConfig<TVoteAccount>,
    ): GetVoteAccountsApiResponse<TVoteAccount>;
```

Then add `<TVoteAccount>` everywhere and also `votePubkey?: Base58EncodedAddress<TVoteAccount>`?